manifests/config.pp

```diff
@@ -415,4 +435,15 @@
     require => File['conntrackd-confdir'],
     notify  => Service['conntrackd'],
   }
+
+  if $systemd {
+  # configuration file
+    file { 'conntrackd-systemd-unit':
+      ensure  => file,
```

Comment:
This parameter could be set by using somehow this other parameter `$conntrackd::ensure` .
Therefore, try to clean as much as possible when ensure is set to absent.